lib/facilities/ppms_client.rb

```diff
@@ -44,6 +44,26 @@ def provider_services(identifier)
       response.body
     end
 
+    def fetch_token
+      token = Redis.current.get('ppms-token')
+      return token if token
+      azure_client = Faraday.new("https://#{Settings.ppms.authority}")
```

Comment:
Realize it's a simple call to fetch a token for use in the PPMS client but we end up with custom Faraday code that uses a different pattern than all our other http calls so I think it implement a client of its own. A custom Faraday call also means it doesn't automatically get wired up to our monitoring and default error handling. What would happen if this call failed? A spec using a VCR cassette that returns a 500 would be one way to find out 😄